src/App.tsx

```diff
@@ -15,36 +17,112 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  ALPHABETIC = 'alphabet',
+  LENGTH = 'length',
+}
+
+type SortFunction = (a: string, b: string) => number;
+
+const sortFunctions: Record<SortType, SortFunction> = {
+  [SortType.ALPHABETIC]: (a, b) => a.localeCompare(b),
+  [SortType.LENGTH]: (good1, good2) => {
+    if (good1.length === good2.length) {
+      return good1.localeCompare(good2);
+    }
+
+    return good1.length - good2.length;
+  },
+};
+
+const getGoods = (
+  goods: string[],
+  sortType: '' | SortType,
+  isReversed: boolean,
+): string[] => {
+  const visibleGoods: string[] = [...goods];
+
+  switch (sortType) {
+    case SortType.ALPHABETIC:
+    case SortType.LENGTH:
+      visibleGoods.sort(sortFunctions[sortType]);
+      break;
+    default:
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+};
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState<SortType | ''>('');
```

Comment:
```suggestion
  const [sortType, setSortType] = useState<SortType>(SortType.Default);
```

Reply:
but i do not have Default, even if i will add it in enum SortType it shows me mistake

Reply:
> but i do not have Default, even if i will add it in enum SortType it shows me mistake

Ok. If you have an error in `sortFunctions` object, you omit this value from your enum with `Exclude` utility type.
https://www.typescriptlang.org/docs/handbook/utility-types.html#excludeuniontype-excludedmembers
https://stackoverflow.com/questions/48850720/typescript-how-can-omit-some-items-from-an-enum-in-typescript